sdk/formrecognizer/azure-ai-formrecognizer/migration-guide.md

```diff
@@ -0,0 +1,255 @@
+
+To improve the development experience across Azure services, a set of uniform [design guidelines][Guidelines] was created for all languages to drive a
+consistent experience with established API patterns for all services. A set of [Java design guidelines][GuidelinesJava] was introduced to ensure that Java clients have a natural and idiomatic feel with respect to the Java ecosystem. Further details are available in the guidelines
+for those interested.
+
+Aside from the benefits of the new design mentioned above, the `azure-ai-formrecognizer (4.0.0-beta.1)` replaces the multiple API design and adds a unified method to analyze documents with the `beginAnalyzeDocument` method and managing of operations in your form recognizer account.This version of the client library defaults to the 2021-09-30-preview version of the service. Refer to the [README][README] for more information about analyzing documents.
+
+### Cross Service SDK improvements
+
+The modern Form Recognizer client library also provides the ability to share in some cross-service improvements made to the Azure development experience, such as
+
+- A unified method for extracting text and structured data from your documents across each of the client libraries.
+- A unified return type `DocumentModel` indicating the document type the model can analyze and the specific fields it can analyze along with the estimated confidence for each field.
+- Specifying a modelId instead of the generated GUID when creating custom models, along with an optional description.
+- Modified `Generate Copy Authorization operation` response to return the target resource information so that it could be used directly when copying custom models method.
+- List Models operation now returns a paged list of prebuilt in addition to custom models and does not return in-progress and failed models.
+- Added methods for getting/listing operations useful to track the status of model creation/copying operations and any resulting errors.
+
+## Important changes
+
+#### Instantiating clients
+
+In 3.x, the `FormRecognizerClient`, instantiated via the `FormRecognizerClientBuilder`. The client contains both sync and async methods.
+In 4.x, the `FormRecognizerClient` has been replaced by the `DocumentAnalysisClient`and the instantiation of the client is done through the [DocumentAnalysisClientBuilder][DocumentAnalysisClientBuilder]. The sync and async operations are separated to [DocumentAnalysisClient][DocumentAnalysisClient] and [DocumentAnalysisAsyncClient][DocumentAnalysisAsyncClient].
+<!-- embedme ./src/samples/java/com/azure/ai/formrecognizer/ReadmeSamples.java#L52-L55 -->
+```java
+DocumentAnalysisClient documentAnalysisClient = new DocumentAnalysisClientBuilder()
```

Comment:
other migration guides I've seen show the old way of doing things and then have a snippet showing the new way. I know it adds a lot to the guide, but it's a nice way to compare.